linera-faucet/server/src/tests.rs

```diff
@@ -134,6 +138,61 @@ async fn test_faucet_rate_limiting() {
     assert_eq!(context.lock().await.update_calls, 4); // Also called in the last error case.
 }
 
+#[tokio::test]
+async fn test_faucet_persistence() {
+    let storage_builder = MemoryStorageBuilder::default();
+    let keys = InMemorySigner::new(None);
+    let clock = storage_builder.clock().clone();
+    clock.set(Timestamp::from(0));
+    let mut builder = TestBuilder::new(storage_builder, 4, 1, keys).await.unwrap();
+    let client = builder
+        .add_root_chain(1, Amount::from_tokens(10))
+        .await
+        .unwrap();
+    let chain_id = client.chain_id();
+    let context = ClientContext {
+        client,
+        update_calls: 0,
+    };
+    let context = Arc::new(Mutex::new(context));
+
+    let faucet_storage = Arc::new(Mutex::new(FaucetStorage::default()));
+    let storage_path = PathBuf::from("/tmp/test_faucet_persistence.json");
```

Comment:
We have `tempfile` crate that manages these temeporary files/dirs for testing.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/4301/commits/06710e573f88e6ef240aa46adcfd62f1c4e59b85.